samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -578,6 +603,11 @@ CHIP_ERROR BLEConnectivityManager::StopScan()
 		return System::MapErrorZephyr(err);
 	}
 
+	if (mScanDoneCallback != ReScanCallback) {
+		/* Scanning has been finished, disable the Scanning state */
+		Instance().UpdateStateFlag(State::Scanning, false);
```

Comment:
Shouldn't we disable Scanning in the scan callback? I think this is the moment that actually scanning is over.

Reply:
But, the scanning can be disabled before the scan callback is called, and what in that situation? Moreover, scanning is over when we call `bt_scan_stop`, not in the scan callback, so I guess the `StopScan` method is the best option to disable the `Scanning` state..

Reply:
> But, the scanning can be disabled before the scan callback is called, and what in that situation?

Right, I actually mean we should disable it in both cases. 

> Moreover, scanning is over when we call bt_scan_stop, not in the scan callback

I don't think so, it's over when it was stopped manually or the scan timeout was reached, so we call `mScanDoneCallback`. I think we proactively always call `StopScan` in the `Connect` method just to not care if the scan is in progress or not.